components/linkedin/linkedin.app.mjs

```diff
@@ -1,11 +1,158 @@
+import { axios } from "@pipedream/platform";
+import constants from "./common/constants.mjs";
+
 export default {
   type: "app",
   app: "linkedin",
-  propDefinitions: {},
+  propDefinitions: {
+    text: {
+      type: "string",
+      label: "Text",
+      description: "Text to be posted on LinkedIn timeline",
+    },
+    organizationId: {
+      type: "string",
+      label: "Organization Id",
+      description: "ID of the organization that will author the post",
+    },
+    type: {
+      type: "string",
+      label: "Type",
+      description: "Type of the post",
+      options: [
+        {
+          label: "Text/URL",
+          value: "NONE",
+        },
+        {
+          label: "Article",
+          value: "ARTICLE",
+        },
+      ],
+      optional: false,
+    },
+    visibility: {
+      type: "string",
+      label: "Visibility",
+      description: "Visibility restrictions on content",
+      options: [
+        {
+          label: "Connections",
+          value: "CONNECTIONS",
+        },
+        {
+          label: "Public",
+          value: "PUBLIC",
+        },
+        {
+          label: "Logged in",
+          value: "LOGGED_IN",
+        },
+      ],
+      optional: false,
+    },
+    postId: {
```

Comment:
Again, `postId` prop can have `async options`, so that users can select from posts.
Again pagination might be good here too.

Reply:
the linkedin app doesn't have access permission to retrieve the posts, I talked to Danny and it was decided that for now this field will stay as a text field

Reply:
> Again, `postId` prop can have `async options`, so that users can select from posts. Again pagination might be good here too.

the linkedin app doesn't have access permission to retrieve the posts, I talked to Danny and it was decided that for now this field will stay as a text field

Reply:
I will add to this that in general it's counterintuitive to include async options on a **delete** action for the object being deleted.
Presumably once it's been deleted it wouldn't show up in the async options 😄